### PR TITLE
Add support for creating internal networks

### PR DESCRIPTION
**Pull request by jgiannuzzi**
Signed-off-by: Jonathan Giannuzzi <jonathan@giannuzzi.be>

### CODE REVIEW

**Comment by jgiannuzzi** [comment]
This should also fix #1040 and #1041.

File: docker/api/network.py [hunk 1]
```diff
@@ -33,6 +33,13 @@ def create_network(self, name, driver=None, options=None, ipam=None,
             'IPAM': ipam,
             'CheckDuplicate': check_duplicate
         }
+
+        if internal:
+            if version_lt(self._version, '1.22'):
```
Comment (jgiannuzzi) [inline docker/api/network.py]: Creating an internal network requires API version 1.22

File: tests/integration/network_test.py [hunk 2]
```diff
@@ -298,3 +298,9 @@ def test_connect_with_ipv6_address(self):
         self.assertEqual(
             net_data['IPAMConfig']['IPv6Address'], '2001:389::f00d'
         )
+
+    @requires_api_version('1.23')
```
Comment (jgiannuzzi) [inline tests/integration/network_test.py]: Checking whether a network is internal requires API version 1.23

**Comment by shin-** [comment]
Changes LGTM. Thanks!

**Comment by jgiannuzzi** [comment]
I am preparing a pull request for docker-compose to add support for creating internal networks.
In order to successfully run the test suite, I need to make it depend on the latest git version of docker-py.
What would be the best way to submit that pull request? Depend on docker-py==1.9.0, even though it has not yet been released?

**Comment by shin-** [comment]
@jgiannuzzi : You should use a git sha to make sure the branch remains functional and can be tested. See for example https://github.com/docker/compose/commit/85c7d3e5ce821c7e8d6a7c85fc0b786f3a60ec93

**Comment by jgiannuzzi** [comment]
Thanks @shin-!
Submitted PR https://github.com/docker/compose/pull/3488

**Pull request merged**